app/controllers/rentals_controller.rb

```diff
@@ -76,11 +76,16 @@ def create
 
     @start_date = params['start_date'] || Time.zone.today
 
-    if @rental.save
+    if @rental.create_reservation && @rental.save
       flash[:success] = 'You have succesfully reserved your Rental!'
       redirect_to(@rental)
     else # error has problem, cannot rental a error message here
-      @rental.errors.full_messages.each { |e| flash_message :warning, e, :now }
+      set_users
+      if @rental.item_type_id.present? && Rental.where(item_type_id: @rental.item_type_id).count == Item.where(item_type_id: @rental.item_type_id).count
```

Comment:
1. Perhaps you can shorten the syntax a bit: `if Rental.where(item_type: @rental.item_type).count == Item.where(item_type: @rental.item_type).count`. If Rentals or Items have a `nil` item_type, we are going to give ourselves headaches :-p
2. I am not sure this logic works without considering rentals over time. That is, rentals in the past, present, and future.